### ISSUE TITLE
"Not Logged In" error on Login

### ISSUE THREAD

**Issue by KerryRitter** [jeduan/cordova-plugin-facebook4#258]
Platform: Android
Plugin Version: 1.7.1

**Facebook Android Platform Settings**
Package Name: com.ionicframework.mbcapp807430
Class Name: com.ionicframework.mbcapp807430.MainActivity

AndroidManifest.xml:
`
<?xml version='1.0' encoding='utf-8'?>
<manifest android:hardwareAccelerated="true" android:versionCode="1" android:versionName="0.0.1" package="com.ionicframework.mbcapp807430" xmlns:android="http://schemas.android.com/apk/res/android">
    <supports-screens android:anyDensity="true" android:largeScreens="true" android:normalScreens="true" android:resizeable="true" android:smallScreens="true" android:xlargeScreens="true" />
    <uses-permission android:name="android.permission.INTERNET" />
    <application android:hardwareAccelerated="true" android:icon="@drawable/icon" android:label="@string/app_name" android:supportsRtl="true">
        <activity android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale" android:label="@string/activity_name" android:launchMode="singleTop" android:name="MainActivity" android:theme="@android:style/Theme.DeviceDefault.NoActionBar" android:windowSoftInputMode="adjustResize">
            <intent-filter android:label="@string/launcher_name">
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        <meta-data android:name="com.facebook.sdk.ApplicationId" android:value="@string/fb_app_id" />
        <meta-data android:name="com.facebook.sdk.ApplicationName" android:value="@string/fb_app_name" />
        <activity android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation" android:label="@string/fb_app_name" android:name="com.facebook.FacebookActivity" android:theme="@android:style/Theme.Translucent.NoTitleBar" />
    </application>
    <uses-sdk android:minSdkVersion="16" android:targetSdkVersion="23" />
</manifest>
`

**Source Code**
I have uploaded the code I'm using to https://github.com/KerryRitter/ionic2-facebook
- https://github.com/KerryRitter/ionic2-facebook/blob/master/app/providers/facebookProvider.ts is a provider/wrapper for the plugin
- https://github.com/KerryRitter/ionic2-facebook/blob/master/app/pages/home/home.ts calls the provider

**Error**
Whenever I call the login method, I get a model with the following message (note, i do not have the FB app installed, i use the browser app. however, I am logged in via the browser app.)

> Not Logged In: You are not logged in. Please login and try again.

Neither the success nor error callbacks seem to get called when this modal opens. I have tried various app settings but I have been unable to resolve the issue. I also tried adding browserInit() to the platform.ready() but to no avail.

**Comment by ammichael** [comment]
I'm having this same issue, but in iOS. At first I thought the problem was iOS 10. But then I tried in iOS 9 and got the same result.

Xcode log:

> 2016-07-07 17:36:49.852 Finances[38523:9961132] Starting login
2016-07-07 17:36:49.854 Finances[38523:9961132] -canOpenURL: failed for URL: "fbauth2:/" - error: "(null)"
2016-07-07 17:36:49.859 Finances[38523:9961132] -canOpenURL: failed for URL: "fbauth2:/" - error: "(null)"
2016-07-07 17:36:49.879 Finances[38523:9961132] THREAD WARNING: ['FacebookConnectPlugin'] took '27.237061' ms. Plugin should use a background thread.
2016-07-07 17:37:20.963 Finances[38523:9961132] ERROR: User cancelled.

**Comment by KerryRitter** [comment]
@ammichael what are your app's platform settings? perhaps we can find a common problem.

**Comment by yoohoogun114** [comment]
Same Problem Here. My platform and Setting is same with @KerryRitter. I Checked My Facebook Developer Setting but not works.

**Comment by ammichael** [comment]
Guys, I'm using ng-cordova-oauth until this is fixed. Is not the best solution, but in my case suits fine.
Maybe for you guys too:

https://github.com/nraboy/ng-cordova-oauth

**Comment by yoohoogun114** [comment]
That's Works? I got "Cannot authenticate via a web browser" error message In Ionic 1.x

**Comment by ammichael** [comment]
Yep, works perfectly fine. But the catch is you need to test in devices/simulator, since it is a cordova plugin..

**Comment by yoohoogun114** [comment]
I was simulated in Android. Unfortunately it didn't works. But Thanks for Feedback.

**Comment by jrtorre** [comment]
Back to facebook login when login is ok, not returns to app in iOS

**Comment by caioavidal** [comment]
I had the same issue here. Solved using ng-cordova-oauth. Unfortunately it doesn't work with facebook native app. Cordova Oauth always open facebook in "browser mode".

**Comment by WilliamIsted** [comment]
I had the same error message displayed (different error I think though), solved it by adding iOS app to Facebook Developer App Settings with the app Bundle ID.

**Comment by jabas06** [comment]
Platform: iOS 9/10
Plugin Version: 1.7.4

I've the same issue on iOS. I'm getting the message "You are not logged in." when calling the login method. 

I'm using Ionic 2 and the 1.7.4 version of this plugin. I tested it on iOS emulators 9.3 and 10.

I also added the iOS platform and the Bundle ID to my Facebook app settings. 

On android everything works fine.

![screen shot 2016-09-25 at 11 47 58 pm](https://cloud.githubusercontent.com/assets/3453385/18822959/99f475c0-837a-11e6-8e69-7c0e81706ee9.png)
